### PR TITLE
Feature/spectrum with sliders

### PR DESCRIPTION
**Pull request by danielbabiak**

### CODE REVIEW

File: src/Spectre.AngularClient/src/app/heatmaps/shared/heatmap.service.spec.ts [hunk 1]
```diff
@@ -46,5 +46,7 @@ describe('HeatmapService', () => {
         expect(heatmap.data[4][0]).toEqual(1.11);
         expect(heatmap.data[0][2]).toEqual(9.99);
       });
+
+      //TODO: test min column and min row
```
Comment (gmrukwa) [inline src/Spectre.AngularClient/src/app/heatmaps/shared/heatmap.service.spec.ts]: We should merge tested code. Please, provide it before the merge.

File: src/Spectre.AngularClient/src/app/preparations/preparation/preparation.component.html [hunk 2]
```diff
@@ -5,9 +5,18 @@
...
+<div>
+  <div class="inlineItemsContainer">
+    <md-slider class="sliderY" [(ngModel)]="yCoordinate" showTicks="true" min="0" max={{yHeatmapSize}} step="1" vertical="true" thumb-label="true" tick-interval="true"
+           (change)="onChangedYCoordinate($event)"></md-slider>
+
+    <app-heatmap
+      [heatmapData]=heatmapData>
+    </app-heatmap>
+  </div>
+  <md-slider class="sliderX" [(ngModel)]="xCoordinate" showTicks="true" min="0" max={{xHeatmapSize}} step="1" vertical="false" thumb-label="true" tick-interval="true"
```
Comment (gmrukwa) [inline src/Spectre.AngularClient/src/app/preparations/preparation/preparation.component.html]: I'd go with this over the heatmap, and `channelId` slider below the heatmap, so it is just over the spectrum.

File: src/Spectre.AngularClient/src/app/preparations/preparation/preparation.component.html [hunk 3]
```diff
@@ -5,9 +5,18 @@
 <md-slider class="slider" [(ngModel)]="currentChannelId" showTicks="true" min="0" max={{mzLenth}} step="1" (input)="onInputChannelId($event)"
            (change)="onChangedChannelId($event)" vertical="false" thumb-label="true" tick-interval="true"></md-slider>
 
-<app-heatmap
-  [heatmapData]=heatmapData>
-</app-heatmap>
+<div>
+  <div class="inlineItemsContainer">
+    <md-slider class="sliderY" [(ngModel)]="yCoordinate" showTicks="true" min="0" max={{yHeatmapSize}} step="1" vertical="true" thumb-label="true" tick-interval="true"
```
Comment (gmrukwa) [inline src/Spectre.AngularClient/src/app/preparations/preparation/preparation.component.html]: Can we make properties in consecutive lines? How will we define standard for this? I think it would be more readable.

File: src/Spectre.AngularClient/src/app/preparations/preparation/preparation.component.ts [hunk 4]
```diff
@@ -77,13 +83,31 @@ export class PreparationComponent implements OnInit {
...
+    this.getSpectrumByCoordianates();
+  }
+
   getSpectrum(selectNumber: number) {
     this.spectrumService
       .get(this.id, selectNumber)
       .subscribe(spectrum => this.spectrumData = this.toSpectrumDataset(spectrum));
   }
 
+  getSpectrumByCoordianates() {
```
Comment (gmrukwa) [inline src/Spectre.AngularClient/src/app/preparations/preparation/preparation.component.ts]: Typo.

File: src/Spectre.AngularClient/src/app/preparations/preparation/preparation.component.ts [hunk 5]
```diff
@@ -77,13 +83,31 @@ export class PreparationComponent implements OnInit {
...
+    this.spectrumService
+      .getByCoordinates(this.id, this.xCoordinate + this.minHeatmapColumn, (this.yCoordinate - this.yHeatmapSize)*(-1) + this.minHeatmapRow)
+      .subscribe(spectrum => this.spectrumData = this.toSpectrumDataset(spectrum), error => this.showError("Spectrum not found"));
```
Comment (gmrukwa) [inline src/Spectre.AngularClient/src/app/preparations/preparation/preparation.component.ts]: The `error` path is just added as a string somewhere in the page. I think we should either go for popup, or e.g. change colors and disable spectrum plot.

File: src/Spectre.AngularClient/src/app/spectra/shared/spectrum.service.ts [hunk 6]
```diff
@@ -53,5 +61,9 @@ function toSpectrum(response: Response): Spectrum {
     intensities: json.Intensities,
     x: json.X,
     y: json.Y
-  });
+  })
 }
+
```
Comment (gmrukwa) [inline src/Spectre.AngularClient/src/app/spectra/shared/spectrum.service.ts]: Leave just one trailing empty line.

File: src/Spectre/stylecop.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
+﻿{
```
Comment (gmrukwa) [inline src/Spectre/stylecop.json]: Why did this file appear here?

File: src/Spectre/Spectre.csproj [hunk 8]
```diff
@@ -315,6 +316,7 @@
     <Analyzer Include="..\packages\StyleCop.Analyzers.1.0.2\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll" />
     <Analyzer Include="..\packages\StyleCop.Analyzers.1.0.2\analyzers\dotnet\cs\StyleCop.Analyzers.dll" />
   </ItemGroup>
+  <ItemGroup />
```
Comment (gmrukwa) [inline src/Spectre/Spectre.csproj]: Remove this line.

File: src/Spectre/Spectre.csproj [hunk 9]
```diff
@@ -293,6 +293,7 @@
     <None Include="Project_Readme.html" />
     <None Include="Properties\PublishProfiles\Debug.pubxml" />
     <None Include="Properties\PublishProfiles\Release.pubxml" />
+    <Content Include="stylecop.json" />
```
Comment (gmrukwa) [inline src/Spectre/Spectre.csproj]: Why is this included?

**Comment by gmrukwa** [comment]
[Select spectrum by coordinates](https://trello.com/c/t1VSnx81/218-select-spectrum-by-coordinates)

File: src/Spectre.AngularClient/package-lock.json [hunk 10]
```diff
@@ -6531,10 +6531,20 @@
       "resolved": "https://registry.npmjs.org/@types/node/-/node-6.0.85.tgz",
       "integrity": "sha512-6qLZpfQFO/g5Ns2e7RsW6brk0Q6Xzwiw7kVVU/XiQNOiJXSojhX76GP457PBYIsNMH2WfcGgcnZB4awFDHrwpA=="
```
[the remark this answers is not in the document]
Comment (gmrukwa) [reply src/Spectre.AngularClient/package-lock.json]: Details are [here](https://docs.npmjs.com/files/package-lock.json). It points specific versions of packages, along with their sources and checksums. You can `npm install` in the project directory to get *exactly* the same version as rest of the team (`packages.json` is not specific enough in this matter).

**Comment by danielbabiak** [comment]
Bug in codefactor - show duplicated code on the same line
